libevmasm/RuleList.h

```diff
@@ -438,7 +438,7 @@ template <class Pattern>
 std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 	Pattern A,
 	Pattern B,
-	Pattern,
+	Pattern Z,
```

Comment:
Since this is called as ``rules += simplificationRuleListPart7(A, B, C, W, X);``, the third parameter will match constants - if you need another pattern ranging over arbitrary expressions, you need to additionally pass ``Y`` or ``Z`` in here on the call site instead.